### DISCUSSION TITLE
Date picker

### DISCUSSION THREAD

**Discussion by dam-man** [livewire/flux#769] [Feature Requests] [answered]
I think we are still missing a good datepicker.  Is this one planned for upcoming releases?

**Comment by ju5t** [comment]
There's another discussion about [this](https://github.com/livewire/flux/discussions/144) too. Date pickers aren't easy though, especially to keep things accessible, so I can imagine it'll take some time. If I had a say a good date picker would be top priority, but I don't, so we'll have to wait :)

**Reply by emmanuel-at-cleverlake** [reply]
You are right. All apps needs at least one date picker.

**Reply by calebporzio** [reply]
Can you guys expound on what you need and in what context you would use it so that I have more to go off of when I tackle it? 

Thanks!

**Reply by kyranb** [reply]
@calebporzio 

These are by no means requirements, just some features I've found myself using :)

- month & year picker, to be able to jump to a specific month/year quickly.
- an optional time, just makes things a bit clearer than having two separate fields.
- timezone awareness (helpful to show what timezone the date/time is being selected in, particularly if not the users local timezone).
- ranges (having a start and end date)
- restricted dates/min and max dates. Think not being able to view a report older than the date an account was created. Or not being able to book a hotel on the nights that it's full.
- one click pre-filled ranges, Thhink 1 day, 7 days, 30 days, this year, last year etc as found in Google Analytics/Stripe's dashboards.
- Amount of months to show at once (one, two or three etc). Helpful for when selecting a range of weeks. On the other hand, say when entering a birthday, there's not really a need to show more than one month.

**Reply by joshhanley** [reply]
@kyranb yep those cover all the use-cases I can think of that I've used before too. I could certainly understand if time wasn't included initially to simplify the component (as then you don't need to worry about timezones, etc).

**Reply by ju5t** [reply]
We would eventually use most what @kyranb mentioned. But to start a date picker with easy navigation (e.g. jump to year/month) and access to the underlying javascript object would be sufficient. We need access to the javascript object to display week numbers inside the input field; I don't think this is something many people would need so we can make that ourselves.

About the context: for us it's just simple dates, think of a delivery date for example.

**Reply by dam-man** [reply]
@calebporzio I think what @kyranb wrote fits most of the use cases. But starting with a simple picker as MVP would be a good first version. So picking a date is what we need and then the ranges can come and so on. This way you migth extend it all the time :)

I hope you can do something for us 👍

**Reply by achatzi78** [reply]
@calebporzio One great date picker library that I have used a lot in my older projects is [Date Range Picker](https://www.daterangepicker.com).  It has a lot of features but the one that I have used extensively is the predefined ranges.  You can define buttons with specific ranges (ex 1 year back, last 2 years, etc) which is very helpful to the users.

You can use that as a reference for features.

Thanks.

**Reply by dicktornfeldt** [reply]
What @kyranb listed is like what I would expect in maybe a year. Would be nice with a "simple" MVP, just be able to slap type="date" on the input field and get something that works good enough cross browsers. 

Atm Safari is my nemesis (as always) 
![Screenshot 2024-12-17 at 09 56 32](https://github.com/user-attachments/assets/a825972c-2dd2-4be0-b3bd-2c471a3d9e91)

**Comment by calebporzio** [comment, marked as answer]
Date picker just shipped in Flux v2: https://fluxui.dev/components/date-picker
